pkg/apis/autoscaling/v2beta1/conversion.go

```diff
@@ -289,3 +294,46 @@ func Convert_v2beta1_PodsMetricStatus_To_autoscaling_PodsMetricStatus(in *autosc
 	}
 	return nil
 }
+
+func Convert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutoscaler(in *autoscaling.HorizontalPodAutoscaler, out *autoscalingv2beta1.HorizontalPodAutoscaler, s conversion.Scope) error {
+	if err := autoConvert_autoscaling_HorizontalPodAutoscaler_To_v2beta1_HorizontalPodAutoscaler(in, out, s); err != nil {
+		return err
+	}
+	if in.Spec.Behavior != nil {
+		old := out.Annotations
+		out.Annotations = make(map[string]string, len(old)+1)
+		for k, v := range old {
+			out.Annotations[k] = v
+		}
+	}
+
+	if in.Spec.Behavior != nil {
+		behaviorEnc, err := json.Marshal(in.Spec.Behavior)
+		if err != nil {
+			return err
+		}
+		out.Annotations[autoscaling.BehaviorSpecsAnnotation] = string(behaviorEnc)
```

Comment:
What if this annotation already existed?  This logic needs to handle every pathology you can think up.  It may be aceptable to nuke it (assume ownership), but please document that here.

Reply:
I'm not an expert in this "conversion" mechanics, but I doubt that this might cause any problem. I'd like to ask for a piece of advice here from an expert.
1. Is it possible that this annotation exists? This is a conversion from the most recent "HPA" spec to the v2beta1 spec. The most recent spec contains behavior, and we don't have any code to generate the annotation with behavior in it.
2. Even if the annotation for behavior exists, we will just rewrite it and increase the size of the "out.Annotation"  map.
3. I repeated the "Annotation" generation from here: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/autoscaling/v1/conversion.go#L327

Reply:
Added a comment about possible overwriting the annotation.

Reply:
Can you also mention it in the __User facing changes__ section of the PR description.